rcon/squad_automod/automod.py

```diff
@@ -63,154 +65,151 @@ def is_time(times: List[datetime], interval_seconds: int):
     return True
 
 
+def _get_team_count(team_view, team):
+    return sum(
+        len(s.get("players", [])) for s in team_view[team].get("squads", {}).values()
+    )
+
+
 def should_warn_player(
-    watch_status: WatchStatus, config: NoLeaderConfig, squad_name, player
+    watch_status: WatchStatus,
+    config: NoLeaderConfig,
+    squad_name: str,
+    aplayer: APlayer
 ):
+    warnings = watch_status.warned.setdefault(aplayer.name, [])
+
+    if len(warnings) == 0:
+        logger.debug("Noted but won't warn: %s in %s", aplayer.short_repr(), squad_name)
+        warnings.append(datetime.now())
+        return PunishStepState.WAIT
+
     if config.number_of_warning == 0:
```

Comment:
You should move that before the previous block

Reply:
Did reorder those blocks to have the grace period for all (warn, punish, kick) without implementing it in each method separately.
If there is a more elegant way let me know